ianalyzer_readers/readers/rdf.py

```diff
@@ -0,0 +1,45 @@
+from typing import Iterable
+
+from rdflib import BNode, Graph
+
+from .core import Reader, Document, Source
+
+
+class RDFReader(Reader):
+    '''
+    A base class for Readers of Resource Description Framework files.
+    These could be in Turtle, JSON-LD, RDFXML or other formats,
+    see [rdflib parsers](https://rdflib.readthedocs.io/en/stable/plugin_parsers.html)
+    '''
+
+    def source2dicts(self, source: Source) -> Iterable[Document]:
+        '''
+        Given a RDF source file, returns an iterable of extracted documents.
+
+        Parameters:
+            source: the source file to extract. This is a string of the file path
+
+        Returns:
+            an iterable of document dictionaries. Each of these is a dictionary,
+                where the keys are names of this Reader's `fields`, and the values
+                are based on the extractor of each field.
+        '''
+        filename = source
+        g = Graph()
+        g.parse(filename)
+        document_subjects = self.document_subjects(g)
+        for subject in document_subjects:
+            yield self._document_from_subject(g, subject)
+
+    def document_subjects(self, graph: Graph) -> list:
+        ''' override this function such that each subject can be used to
+        retrieve a separate document in the resulting index
```

Comment:
I had to read this sentence a couple of times and could not figure out what it meant. The role of this function is clear if you read the implementation of `source2dicts`, but not from this docstring. Since the docstring is instructing the user to write their own code, it should provide clear instructions about what this function should do.

Also, keep in mind that "index" is not a concept in this package.